elastic/eui
# Add basic React 18 support

## Summary

This is the first set of changes needed to make EUI compatible with React 18. Following @cee-chen suggestion in #6772, I divided the work into more branches that are easier to review and merge into our feature branch (`feature/react-18`).

This PR contains:

- Upgrade react, react-dom and its types to v18
- Update package.json resolutions to force @types/react version to v18
- Fix component types to follow React 18 changes, most notably add `PropsWithChildren` to components accepting the `children` property
- Switch to using the new `createRoot` API in docs entrypoint

## QA

1. Pull the changes and run `yarn` to fetch new dependencies
2. Run `yarn build` and confirm the application builds successfully
3. Run `yarn start`, wait for it to load and open `http://localhost:8030` to confirm the application is loading
4. See the homepage and other component pages opening as expected

Please note that the work is not final and there are additional changes that need to be made to make the application stable and tests passing. At this stage, some pages may crash the application, portals are not working and the majority of tests are failing. This is expected and will be addressed in next PRs.

### General checklist
